### PR TITLE
feat(request_reviews): allow to pick random user from teams

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-603

Change-Id: I2e79458662a64bcff928c139ff0fa5c5b73de25e

### CODE REVIEW

**Comment by sileht** [comment]
doc and releasenotes ?

File: mergify_engine/actions/request_reviews.py [hunk 1]
```diff
@@ -165,6 +174,21 @@ async def run(
...
+            else:
+                self.config["users"].update(
+                    {
+                        user: weight
+                        for user in await ctxt.repository.installation.get_team_members(
+                            team.team
+                        )
+                    }
+                )
+
```
Comment (jd) [inline mergify_engine/actions/request_reviews.py:191]: This adds all users from a single team. I doubt anyone wants that.
It's likely people wants a subset of N people from a team.

See https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-assignment-for-your-team for what GitHub supports already.

**Review by jd — DISMISSED** [review]

File: mergify_engine/actions/request_reviews.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/actions/request_reviews.py:191]: `users_from_teams` without settings `random_count` doesn't make sense.

**Comment by sileht** [comment]
@Mergifyio squash

File: docs/source/actions/request_reviews.rst [hunk 2]
```diff
@@ -35,6 +35,11 @@ Options
     - list of string or dictionary of login and weight
     -
     - The team name to request reviews from.
+  * - ``users_from_teams``
+    - list of string or dictionary of login and weight
+    -
+    - The team names to get the list of users to request reviews from.
+
```
Comment (jd) [inline docs/source/actions/request_reviews.rst:42]:
```suggestion
```

File: mergify_engine/actions/request_reviews.py [hunk 3]
```diff
@@ -75,6 +75,15 @@ class RequestReviewsAction(actions.Action):
...
             },
         ),
+        voluptuous.Required("users_from_teams", default=list): voluptuous.All(
+            voluptuous.Any(
+                [types.GitHubTeam],
+                {
+                    types.GitHubTeam: _random_weight,
+                },
+            ),
+            voluptuous.Coerce(_ensure_weight),
```
Comment (jd) [inline mergify_engine/actions/request_reviews.py:85]: coercing too much

**Review by jd — CHANGES_REQUESTED** [review]

**Review by jd — APPROVED** [review]

**Pull request merged**